tests/test_client.py

```diff
@@ -58,6 +66,50 @@ def test_correct_new_auth0(self):
             auth0=new_auth0,
         )
 
+    @pytest.mark.parametrize(
+        "env, core_endpoint_expected",
+        [
+            ('prod', 'https://apis.pasqal.cloud/core-fast'),
+            ('preprod', 'https://apis.preprod.pasqal.cloud/core-fast'),
+            ('dev', 'https://apis.dev.pasqal.cloud/core-fast')
+        ]
+    )
+    def test_good_env(self, env, core_endpoint_expected):
+        sdk = SDK(
+            project_id=self.project_id,
+            username=self.username,
+            password=self.password,
+            env=env
+        )
+        assert sdk._client.endpoints.core == core_endpoint_expected
+
+    def test_bad_env(self):
+        with pytest.raises(ValueError, match="This environment does not exist, valid options are"):
+            SDK(
+                project_id=self.project_id,
+                username=self.username,
+                password=self.password,
+                env="wrong"
+            )
+
+    @pytest.mark.filterwarnings(
+        "ignore:The parameters 'endpoints' and 'auth0' are deprecated, from now use 'env' instead"
+    )
+    def test_endpoints_auth0_env_given(self):
```

Comment:
Can you add several test: case where user inputs:
- env + auth0 
- env + endpoints
- env + auth0 + endpoints